test/distributed/test_c10d_gloo.py

```diff
@@ -1483,17 +1477,18 @@ def setUp(self):
 
     def _get_process_group(self):
         store = self._get_store()
-        return c10d.ProcessGroupGloo(store, self.rank, self.world_size)
+        c10d.init_process_group(backend="gloo", store=store, rank=self.rank, world_size=self.world_size)
+        return c10d.distributed_c10d._get_default_group()
```

Comment:
`init_process_group` creates a new process group every time. In the case of our tests, we usually have `setUp()` and `tearDown()` to create and destroy PGs for every test

Reply:
I was asking because `self._get_process_group()` is called by most tests in this file. If we can create process group only once, that would greatly speed up the test.

And re `init_process_group`, I think a second call should not create a new PG -- there shouldn't be a second WORLD than the WORLD :) It should just do nothing.

Reply:
Yea, i agree theres some low hanging fruit we can do to speed up tests by not creating new processes / process groups. I think that was one of the selling points of adopting `multi-threaded pgs`. For tests right now, we set up a new PG each time to ensure that each test is isolated and does not effect each other.

For the second point, i think `init_process_group` makes sense to create a new WORLD pg. For the case the user does something like this:
```
init_process_group("nccl", ...)
# do something
init_process_group("gloo", ...)
# it would be expected that you are operating on gloo PG now, not NCCL
```